packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -109,9 +109,13 @@ library LibProposing {
             meta.l1Hash = blockhash(block.number - 1);
             meta.timestamp = uint64(block.timestamp);
 
-            // if multiple L2 blocks included in the same L1 block,
-            // their block.mixHash fields for randomness will be the same.
-            meta.mixHash = bytes32(block.difficulty);
+            // After The Merge, L1 mixHash contains the prevrandao
+            // from the beacon chain. Since multiple Taiko blocks
+            // can be proposed in one Ethereum block, we need to
+            // add salt to this random number as L2 mixHash
+            meta.mixHash = keccak256(
+                abi.encodePacked(block.prevrandao, state.nextBlockId)
```

Comment:
One more question here: do we need to change all the solidity files headers from `pragma solidity ^0.8.9;` to `pragma solidity ^0.8.18;`? since we are actually use the new method which is introduced in 0.8.18.

Reply:
I actually have a separate PR for this